workshops/meme-generator/README.md

```diff
@@ -0,0 +1,385 @@
+  });
+
+  // Bottom text font size
+  fontSize = canvas.width * bottomTextSize;
+  ctx.font = `${fontSize}px Impact`;
+  ctx.lineWidth = fontSize / 20;
+
+  // Draw bottom text
+  ctx.textBaseline = 'bottom';
+  bottomText.split('\n').reverse().forEach((t, i) => { // .reverse() because it's drawing the bottom text from the bottom up
+    ctx.fillText(t, canvas.width / 2, canvas.height - i * fontSize, canvas.width);
+    ctx.strokeText(t, canvas.width / 2, canvas.height - i * fontSize, canvas.width);
+  });
+}
+
+window.addEventListener('DOMContentLoaded', () => {
+  // Initialize variables
+  const topTextInput = document.getElementById('top-text');
+  const bottomTextInput = document.getElementById('bottom-text');
+  const topTextSizeInput = document.getElementById('top-text-size-input');
+  const bottomTextSizeInput = document.getElementById('bottom-text-size-input');
+  const imageInput = document.getElementById('image-input');
+  const generateBtn = document.getElementById('generate-btn');
+  // Default/Demo text
+  topTextInput.value = 'Top\nValue';
+  bottomTextInput.value = 'Bottom\nValue';
+
+  // Generate button click listener
+  generateBtn.addEventListener('click', () => {
+    // Read image as DataURL using the FileReader API
+    const reader = new FileReader();
+    reader.onload = () => {
+      const img = new Image();
+      img.src = reader.result;
+      img.onload = () => {
+        generateMeme(img, topTextInput.value, bottomTextInput.value, topTextSizeInput.value, bottomTextSizeInput.value);
+      };
+    };
+    reader.readAsDataURL(imageInput.files[0]);
+  });
+});
+
+```
+
+</summary>
+</details>
+
+**_And with this, we finish coding our Meme-Genie, Check out what you've just built by clicking the "Run" Button on top!_**
+
+![Final Output](https://cloud-8kdhhtdj0.vercel.app/0ezgif.com-gif-maker.gif)
+
+>Note: To save the generated meme, you can right click the meme and click "save as...", to save the meme!
+
+## Part 4: Hacking
+
+Now that we have finished building a very basic website, we can add a lot of cool things to it!
+
+Here are some things you can try to add:
+
+1. Try to add [colour picker](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/color), so that user can change topText/bottomText colour.
+2. Add a drag and drop feature, using the [HTML5 Drag and Drop API](https://web.dev/drag-and-drop/).
+3. Make a [Flexbox](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Flexbox) to create a nice layout.
+4. Use an API to fetch meme-templates, like the one provided by [ImgFlip](https://imgflip.com/api).
+5. Add Share and Download buttons, to share your super awesome meme!
+
+## Part 5: The End
+
+If you haven't created an account on [repl.it](https://repl.it), make sure you do so to save this wonderful piece of creation!
+
+If you still face difficulties in signing up watch [this](https://www.youtube.com/watch?v=Mtqp4CUepk0).
```

Comment:
I don't think you should include this. Maybe instead say "If you're having trouble signing up, ask your club leader" or something like that